### PR TITLE
Fix mapping of default localization resources

### PR DESCRIPTION
**Pull request by danyf90**
Resolves https://github.com/tuist/tuist/issues/4435

### Short description 📝

Use string extension instead of lproj  extension

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally.

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift [hunk 1]
```diff
@@ -657,7 +657,7 @@ extension ResourceFileElements {
         "xcdatamodeld",
         "xcmappingmodel",
         "xcassets",
-        "lproj",
+        "strings",
```
Comment (shahzadmajeed) [inline Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:660]: Could we keep both?

Comment (danyf90) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:660]: What do you mean? aren't lproj made of strings files?

Comment (shahzadmajeed) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:660]: True, I thought we are looking at localized directories and files. Never mind this looks good

**Review by shahzadmajeed — APPROVED** [review]

**Comment by danyf90** [comment]
> Hey @danyf90, thank you so much for the amazing work. We really appreciate it. Is this pr ready to be merge or can I help with something?
> 
> Thanks in advance

Looks like tests are failing so something is wrong :D
If anyone wants to have a look at it feel free to do as I don't have time now!

File: projects/tuist/fixtures/app_with_spm_dependencies/App/Project.swift [hunk 2]
```diff
@@ -34,6 +34,7 @@ let project = Project(
                 .external(name: "FirebaseDatabase"),
                 .external(name: "FirebaseFirestore"),
                 .external(name: "IterableSDK"),
+                .external(name: "StripeCore"),
```
Comment (ffittschen) [inline projects/tuist/fixtures/app_with_spm_dependencies/App/Project.swift:37]: `StripeCore` is not a package product. It's an internal target, which is the reason for the failing unit tests

File: projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift [hunk 3]
```diff
@@ -8,6 +8,7 @@ import FirebaseCrashlytics
 import FirebaseDatabase
 import FirebaseFirestore
 import IterableSDK
+import StripeCore
```
Comment (ffittschen) [inline projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift:11]:
```suggestion
import Stripe
```

File: projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift [hunk 4]
```diff
@@ -33,6 +34,9 @@ public enum AppKit {
         // Use FirebaseFirestore to make sure it links fine
         _ = Firestore.firestore()
 
+        // Use Strip to make sure it links fine
+        _ = StripeCore.module
```
Comment (ffittschen) [inline projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift:38]:
```suggestion
        _ = STPAPIClient.shared
```

File: projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift [hunk 5]
```diff
@@ -33,6 +34,9 @@ public enum AppKit {
         // Use FirebaseFirestore to make sure it links fine
         _ = Firestore.firestore()
 
+        // Use Strip to make sure it links fine
```
Comment (ffittschen) [inline projects/tuist/fixtures/app_with_spm_dependencies/App/Sources/AppKit/AppKit.swift:37]:
```suggestion
        // Use Stripe to make sure it links fine
```

File: projects/tuist/fixtures/app_with_spm_dependencies/LocalSwiftPackage/Sources/Styles/Resources/XCMapping.xcmappingmodel [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (danyf90) [reply projects/tuist/fixtures/app_with_spm_dependencies/LocalSwiftPackage/Sources/Styles/Resources/XCMapping.xcmappingmodel]: Removed this for now

File: projects/tuist/fixtures/app_with_spm_dependencies/LocalSwiftPackage/Package.swift [hunk 7]
```diff
@@ -9,7 +9,7 @@ let package = Package(
     platforms: [.iOS(.v13)],
     products: [.library(name: "Styles", targets: ["Styles"])],
     targets: [
-        .target(name: "Styles", resources: [.process("Resources")]),
```
[the remark this answers is not in the document]
Comment (danyf90) [reply projects/tuist/fixtures/app_with_spm_dependencies/LocalSwiftPackage/Package.swift:12]: Thanks, added back resources for Fonts only

**Review by unxavi — APPROVED** [review]

**Review by unxavi — APPROVED** [review]

**Review by jesus-mg-ios — APPROVED** [review]

**Review by luispadron — APPROVED** [review]

**Pull request merged**